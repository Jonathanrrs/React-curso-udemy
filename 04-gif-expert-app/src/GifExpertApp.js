import React, {useState} from 'react';

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Dragon ball', 'Naruto'];

    const [categories, setCategories] = useState(['One Punch', 'Dragon ball', 'Naruto']);

    const hanleAdd = () => {
        setCategories([...categories, 'Death Note']);
    }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />

           
            <ol>
                {
                    categories.map((category) => {
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp;