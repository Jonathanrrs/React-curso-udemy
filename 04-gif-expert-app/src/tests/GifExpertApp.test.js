import React from 'react';
import {shallow} from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Comprobar el componente GifExpertApp', () => {
    

    test('Debe mostrar el componente correctamente ', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['Naruto', 'Dragon ball'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifsGrid').length).toBe(categories.length);

    })
    

})
