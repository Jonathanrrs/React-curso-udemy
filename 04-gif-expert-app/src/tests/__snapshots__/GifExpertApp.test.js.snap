// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comprobar el componente GifExpertApp Debe mostrar el componente correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Comprobar el componente GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifsGrid
      category="Naruto"
      key="Naruto"
    />
    <GifsGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
